service/github/check.go

```diff
@@ -0,0 +1,354 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/go-github/v60/github"
+	"github.com/reviewdog/reviewdog"
+	"github.com/reviewdog/reviewdog/cienv"
+	"github.com/reviewdog/reviewdog/filter"
+	"github.com/reviewdog/reviewdog/proto/rdf"
+	"github.com/reviewdog/reviewdog/service/github/githubutils"
+)
+
+// GitHub check runs API cannot handle too large requests.
+// Set max number of filtered findings to be shown in check-run summary.
+// ERROR:
+//
+//	https://api.github.com/repos/easymotion/vim-easymotion/check-runs: 422
+//	Invalid request.
+//	Only 65535 characters are allowed; 250684 were supplied. []
+const maxAllowedSize = 65535
+
+// > The Checks API limits the number of annotations to a maximum of 50 per API
+// > request.
+// https://developer.github.com/v3/checks/runs/#output-object
+const maxAnnotationsPerRequest = 50
+
+var _ reviewdog.CommentService = (*Check)(nil)
+
+type Check struct {
+	CLI      *github.Client
+	Owner    string
+	Repo     string
+	PR       int // optional
+	SHA      string
+	ToolName string
+	Level    string
+
+	muComments   sync.Mutex
+	postComments []*reviewdog.Comment
+
+	muResult sync.Mutex
+	result   *CheckResult
+}
+
+type CheckResult struct {
+	ReportURL  string
+	Conclusion string
+}
+
+func (ch *Check) Post(_ context.Context, c *reviewdog.Comment) error {
+	ch.muComments.Lock()
+	defer ch.muComments.Unlock()
+	ch.postComments = append(ch.postComments, c)
+	return nil
+}
+
+func (ch *Check) PostFiltered(_ context.Context, c *reviewdog.Comment) error {
+	ch.muComments.Lock()
+	defer ch.muComments.Unlock()
+	ch.postComments = append(ch.postComments, c)
+	return nil
+}
+
+func (ch *Check) GetResult() *CheckResult {
+	ch.muResult.Lock()
+	defer ch.muResult.Unlock()
+	return ch.result
+}
+
+func (ch *Check) Flush(ctx context.Context) error {
```

Comment:
**[golint-circleci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>exported method Check.Flush should have comment or be unexported
<!-- __reviewdog__:ChA4YzM3ZDRiYjRmYWM5ZDYyEg9nb2xpbnQtY2lyY2xlY2k= -->